dev/release/verify-release-candidate.sh

```diff
@@ -51,11 +51,17 @@ HERE=$(cd `dirname "${BASH_SOURCE[0]:-$0}"` && pwd)
 
 ARROW_DIST_URL='https://dist.apache.org/repos/dist/dev/arrow'
 
-: ${ARROW_HAVE_CUDA:=}
-if [ -z "$ARROW_HAVE_CUDA" ]; then
-  if nvidia-smi --list-gpus 2>&1 > /dev/null; then
-    ARROW_HAVE_CUDA=yes
+detect_cuda() {
+  if ! (which nvcc && which nvidia-smi); then
+    return 1
   fi
+
+  local n_gpus=$(nvidia-smi --list-gpus | wc -l)
+  return $((${n_gpus} < 1))
+}
+
+if [ -z "${ARROW_CUDA:-}" ] && detect_cuda; then
```

Comment:
`:-` is needless.

Reply:
Habit of always using `set -u`.

Reply:
We can remove `:-` by moving `: ${ARROW_CUDA:=OFF}` to the above line:

```shell
: ${ARROW_CUDA:=OFF}
if [ -z "${ARROW_CUDA}" ] && detect_cuda; then
```

I think that using `:-` for `set -u` isn't a good idea. If we use `:-` for reference a variable, we can detect typo by `set -u` check.

Reply:
That's a mater of preference. By avoiding setting a default (with `:=`), you can differentiate if the user explicitly defined the variable. But I'll follow the current convention.

Reply:
Ah, sorry. The suggestion I wrote is wrong.